src/spyglass/spikesorting/v1/recording.py

```diff
@@ -202,43 +208,160 @@ def make(self, key):
-                ),
+                "nwb_file_name": nwb_file_name,
                 "interval_list_name": key["recording_id"],
-                "valid_times": sort_interval_valid_times,
+                "valid_times": self._get_sort_interval_valid_times(key),
                 "pipeline": "spikesorting_recording_v1",
-            }
+            },
+            skip_duplicates=True,  # for recompute
         )
-        AnalysisNwbfile().add(
-            (SpikeSortingRecordingSelection & key).fetch1("nwb_file_name"),
-            key["analysis_file_name"],
+        AnalysisNwbfile().add(nwb_file_name, key["analysis_file_name"])
+
+        self.insert1(key)
+        self._record_environment(key)
+
+    def _record_environment(self, key):
+        """Record environment details for this recording."""
+        from spyglass.spikesorting.v1 import recompute as rcp
+
+        rcp.RecordingRecomputeSelection().insert(key, at_creation=True)
+
+    @classmethod
+    def _make_file(
+        cls,
+        key: dict = None,
+        recompute_file_name: str = None,
+        save_to: Union[str, Path] = None,
+        rounding: int = 4,
+    ):
+        """Preprocess recording and write to NWB file.
+
+        All `_make_file` methods should exit early if the file already exists.
+
+        - Get valid times for sort interval from IntervalList
+        - Preprocess recording
+        - Write processed recording to NWB file
+
+        Parameters
+        ----------
+        key : dict
+            primary key of SpikeSortingRecordingSelection table
+        recompute_file_name : str, Optional
+            If specified, recompute this file. Use as resulting file name.
+            If none, generate a new file name. Used for recomputation after
+            typical deletion.
+        save_to : Union[str,Path], Optional
+            Default None, save to analysis directory. If provided, save to
+            specified path. Used for recomputation prior to deletion.
+        """
+        if not key and not recompute_file_name:
+            raise ValueError(
+                "Either key or recompute_file_name must be specified."
+            )
+        if isinstance(key, dict):
+            key = {k: v for k, v in key.items() if k in cls.primary_key}
+
+        hash = None
+        recompute = recompute_file_name and not key and not save_to
+
+        if recompute or save_to:  # if we expect file to exist
+            file_path = AnalysisNwbfile.get_abs_path(
+                recompute_file_name, from_schema=True
+            )
+        if recompute:  # If recompute, check if file exists
+            if Path(file_path).exists():  # No need to recompute
+                return
+            logger.info(f"Recomputing {recompute_file_name}.")
+            query = cls & {"analysis_file_name": recompute_file_name}
+            # Use deleted file's ids and hash for recompute
+            key, recompute_object_id, recompute_electrodes_id, hash = (
```

Comment:
Is it intended that this hash is never used?

Reply:
Thanks for catching this - no, I think this is a holdover from a previous version where I compared hashes here. I've since moved that comparison to `_update_external`. Removing